src/sockperf.cpp

```diff
@@ -1718,7 +1718,27 @@ static int resolve_sockaddr(const char *host, const char *port, int sock_type,
     int res = getaddrinfo(host, port, &hints, &result);
     if (res == 0) {
         get_socket_address(result, addr, addr_len);
+    } else if (res == EAI_NONAME && host != NULL) { // Unix Domain Socket name
+        log_dbg("provided path for Unix Domain Socket is %s\n", host);
+        std::string path = host;
+        if (!(path.rfind("/", 0) == 0)) { // limit to prefix
+            log_err("unix domain socket path have to start with '/'");
+            res = SOCKPERF_ERR_SOCKET;
+            return res;
+        }
+        if (path.length() > MAX_UDS_NAME) {
+            log_err("length of name is larger than %d bytes", MAX_UDS_NAME);
+            res = SOCKPERF_ERR_SOCKET;
+            return res;
+        }
+        addr_len = path.length() + 2;
+        strncpy(((sockaddr_store_t*)addr)->addr_un.sun_path, path.c_str(), addr_len);
```

Comment:
1. I don't understant why you adding 2 bytes to string length.
2. It's enough to call `strncpy(, , MAX_UDS_NAME)`. This handles corner case when `path.length() == MAX_UDS_NAME` so trailing zero is not needed to be copied.